grant-project/grant-monorepo
# User Authentication UI (Pt 3 - Persistence)

### What This Does

Once you log in, you stay logged in. This PR does the following:

* Adds `redux-persist` to the project
    * Each `module` will optionally have a `persistence.ts` file that defines how it's persisted, and in the future, will store the migration code as needed.
* Enables `redux-saga`s, uses a saga to auto-login user
    * It'd be good to circle back around to other reducers and saga-fy them, namely web3

### Things to Improve

* Token isn't actually signed, so its being stored doesn't actually matter. We also store `tokenAddress` right now, but that's just while this is faked.
* It would be cool if the SSR redux store pre-fetched the `user` key if the user is authed and sends a token with the request.

### Steps to Test

* Sign in or sign up
* Refresh page
* Confirm you are still signed in
